rx: Update rx gcc from Renesas release

duynguyenxa opened the pull request:
This commit pickup the changes from GCC for RX release by Renesas to support newer RX devices and functionality
Changes are pickup from https://llvm-gcc-renesas.com/rx/rx-latest-source-code/

stephanosio commented on gcc/config/sh/sh.md:
Are the changes to the `sh` arch bits relevant/necessary?

duynguyenxa replied:
Oopps, sorry this is my bad, seem like this file was accidentally updated because of my editor config for trimming whitespace. I've pushed new commit to remove unnecessary changes

duynguyenxa committed:
rx: Update rx gcc from Renesas release

This commit pickup the changes from GCC for RX release by Renesas
to support newer RX devices and functionality
Changes are pickup from https://llvm-gcc-renesas.com/rx/rx-latest-source-code/

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>
